libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -166,51 +168,73 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
 AP_ADSB::AP_ADSB()
 {
     AP_Param::setup_object_defaults(this, var_info);
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
     if (_singleton != nullptr) {
         AP_HAL::panic("AP_ADSB must be singleton");
     }
+#endif
     _singleton = this;
+
+#ifdef ADSB_STATIC_CALLSIGN
+    memcpy(&out_state.cfg.callsign, ADSB_STATIC_CALLSIGN, 8);
+#endif
 }
 
 /*
  * Initialize variables and allocate memory for array
  */
 void AP_ADSB::init(void)
 {
-    // in_state
-    in_state.vehicle_count = 0;
     if (in_state.vehicle_list == nullptr) {
-        if (in_state.list_size_param != constrain_int16(in_state.list_size_param, 1, ADSB_VEHICLE_LIST_SIZE_MAX)) {
-            in_state.list_size_param.set_and_notify(ADSB_VEHICLE_LIST_SIZE_DEFAULT);
-            in_state.list_size_param.save();
-        }
-        in_state.list_size = in_state.list_size_param;
-        in_state.vehicle_list = new adsb_vehicle_t[in_state.list_size];
+        // sanity check param
+        in_state.list_size_param = constrain_int16(in_state.list_size_param, 1, ADSB_VEHICLE_LIST_SIZE_MAX);
+
+        in_state.list_size_allocated = in_state.list_size_param;
+        in_state.vehicle_list = new adsb_vehicle_t[in_state.list_size_allocated];
 
         if (in_state.vehicle_list == nullptr) {
-            // dynamic RAM allocation of _vehicle_list[] failed, disable gracefully
-            hal.console->printf("Unable to initialize ADS-B vehicle list\n");
-            _enabled.set_and_notify(0);
+            // dynamic RAM allocation of in_state.vehicle_list[] failed, disable gracefully
+            hal.console->printf("Unable to initialize ADSB vehicle list\n");
+            // disable ourselves to avoid repeated allocation attempts
+            _enabled.set(0);
```

Comment:
As this exists in init(), I don't think this code can get called multiple times?

Reply:
correct, but I talked to @tridge about it and it's getting a new init_failed flag so we don't actually disable the param. It will also send a GCS send_text instead this way